Chemistry/src/DataModel/OrbitalIntegral.cs

```diff
@@ -207,22 +207,37 @@ public SpinOrbital[][] EnumerateSpinOrbitals()
                 {
                     var fst = idxOrbital;
                     var lst = OrbitalIndices.Length - idxOrbital - 1;
-                    var spin = (idx / nSpins.Pow(idxOrbital)) % nSpins;
-                    spinOrbitalArray[fst] = new SpinOrbital
-                    {
-                        orbital = OrbitalIndices[fst],
-                        spin = spin
-                    };
-                    spinOrbitalArray[lst] = new SpinOrbital
-                    {
-                        orbital = OrbitalIndices[lst],
-                        spin = spin
-                    };
+                    var spin = (Spin)((idx / nSpins.Pow(idxOrbital)) % nSpins);
+                    spinOrbitalArray[fst] = new SpinOrbital(OrbitalIndices[fst], spin, indexConvention);
+                    spinOrbitalArray[lst] = new SpinOrbital(OrbitalIndices[lst], spin, indexConvention);
                 }
                 spinOrbitalArrayOfArray[idx] = spinOrbitalArray;
             }
             return spinOrbitalArrayOfArray;
         }
+
+        public bool Equals(OrbitalIntegral x)
```

Comment:
```suggestion
        public override bool Equals(object x)
```

You'll likely want something like `if (x is OrbitalIntegral integral) { /* do something with integral here */ } else { return false; }` as well.

Reply:
Fixed